arch/cortex-m/src/syscall.rs

```diff
@@ -232,9 +232,9 @@ impl<A: CortexMVariant> kernel::syscall::UserspaceKernelBoundary for SysCall<A>
         //  - Stack offset 4 is R12, which the syscall interface ignores
         let stack_bottom = state.psp as *mut usize;
         ptr::write(stack_bottom.offset(7), state.psr); //......... -> APSR
-        ptr::write(stack_bottom.offset(6), callback.pc | 1); //... -> PC
+        ptr::write(stack_bottom.offset(6), usize::from(callback.pc) | 1); //... -> PC
```

Comment:
Small style nit, does `callback.pc.into() | 1` (or maybe `callback.pc.into::<usize>() | 1`) not work here? Would better mirror line 237.

Reply:
`callback.pc.into() | 1` type inference falls over (without assigning to a temp or some such), or I would have indeed liked that one.

`into` is not a generic fn, it a method on a generic trait, so the explicit spelling would be:

`<CapabilityPtr as Into<T>>::into(callback.pc)`

Which I like less than `usize::from(callback.pc)`.